core/indigo-core/molecule/molecule_cdxml_loader.h

```diff
@@ -0,0 +1,204 @@
+/****************************************************************************
+ * Copyright (C) from 2009 to Present EPAM Systems.
+ *
+ * This file is part of Indigo toolkit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+#ifndef __cdxml_loader__
+#define __cdxml_loader__
+
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include <functional>
+
+#include "base_cpp/array.h"
+#include "base_cpp/exception.h"
+#include "elements.h"
+#include "molecule/base_molecule.h"
+#include "molecule/molecule_stereocenter_options.h"
+#include "molecule/query_molecule.h"
+
+typedef unsigned short int UINT16;
+typedef int INT32;
+typedef unsigned int UINT32;
+#include "molecule/CDXConstants.h"
+
+class TiXmlHandle;
+class TiXmlElement;
+class TiXmlNode;
+class TiXmlAttribute;
+
+namespace indigo
+{
+    class Scanner;
+    class Molecule;
+    class QueryMolecule;
+
+    class AutoInt
+    {
+    public:
+        AutoInt() : val(0){};
+
+        AutoInt(int v) : val(v){};
+
+        AutoInt(const std::string& v) : val(std::stoi(v))
+        {}
+
+        operator int() const
+        {
+            return val;
+        }
+
+        operator std::string() const
+        {
+            return std::to_string( val );
+        }
+
+    private:
+        int val;
+    };
+
+
+    struct _ExtConnection
+    {
+        int bond_id;
+        int point_id;
+        int atom_idx;
+    };
+
+    struct CdxmlNode
+    {
+        CdxmlNode() : element(ELEM_C), type( kCDXNodeType_Element ) // Carbon by default
+        {
+        }
+        AutoInt id;
+        std::string label;
+        AutoInt element;
+        Vec3f pos;
+        int type;
```

Comment:
is not initialized by default. (is not assigned in constructor and does not have own default constructor)